jazz_services-handler/index.js

```diff
@@ -24,603 +24,275 @@ const config = require('./components/config.js'); //Import the environment data.
+			.then(result => {
+				payload = result.payload;
+				if (result.interested_event) {
+					return processEvent(payload, configData, authToken);
+				} else {
+					handleProcessedEvents(sequenceNumber, payload);
+				}
+			})
+			.then(result => {
+				handleProcessedEvents(sequenceNumber, payload);
+				return resolve(result);
+			})
+			.catch(err => {
+				handleFailedEvents(sequenceNumber, err.failure_message, payload, err.failure_code);
+				return reject(err);
+			});
+	});
+}
+var checkInterest = function (encodedPayload, sequenceNumber) {
+	return new Promise((resolve, reject) => {
+		var payload = JSON.parse(new Buffer(encodedPayload, 'base64').toString('ascii'));
+		if (payload.Item.EVENT_TYPE && payload.Item.EVENT_TYPE.S) {
+			logger.info("found " + payload.Item.EVENT_TYPE.S + " event with sequence number: " + sequenceNumber);
+			resolve({
+				"interested_event": true,
+				"payload": payload.Item
+			});
+		} 
+	});
+}
+
+var processEvent = function (payload, configData, authToken) {
+	return new Promise((resolve, reject) => {
+		if (!payload.EVENT_NAME.S || !payload.EVENT_STATUS.S) {
+			logger.error("validation error. Either event name or event status is not properly defined.");
+			var err = handleError(failureCodes.PR_ERROR_1.code, "Validation error while processing event for service");
+			return reject(err);
+		}
 
-	function HandleFailedEvents(id, error, event, failureCode) {
-		failedEvents.push({
-			Id: id,
-			DelaySeconds: 0,
-			MessageBody: error,
-			MessageAttributes: {
-				"sequence_id": {
-					DataType: "String",
-					StringValue: id.toString()
-				},
-				"event": {
-					DataType: "String",
-					StringValue: JSON.stringify(event)
-				},
-				"failure_code": {
-					DataType: "String",
-					StringValue: failureCode
-				},
-				"failure_message": {
-					DataType: "String",
-					StringValue: error
+		var svcContext = JSON.parse(payload.SERVICE_CONTEXT.S);
+		var serviceContext = getServiceContext(svcContext);
+		serviceContext.service = payload.SERVICE_NAME.S
+		serviceContext.created_by = payload.USERNAME.S
+		serviceContext.service_id = payload.SERVICE_ID.S
+		var statusResponse = getUpdateServiceStatus(payload, configData);
+
+		if (statusResponse.interested_event) {
+			var inputs = {
+				"TOKEN": authToken,
+				"SERVICE_API_URL": configData.SERVICE_API_URL,
+				"SERVICE_API_RESOURCE": configData.SERVICE_API_RESOURCE,
+				"ID": serviceContext.service_id,
+				"DESCRIPTION": serviceContext.description,
+				"REPOSITORY": serviceContext.repository,
+				"EMAIL": serviceContext.email,
+				"SLACKCHANNEL": serviceContext.slackChannel,
+				"TAGS": serviceContext.tags,
+				"ENDPOINTS": serviceContext.endpoint,
+				"STATUS": statusResponse.status,
+				"METADATA": serviceContext.metadata
+			};
+
+			crud.update(inputs, function (err, results) {
+				if (err) {
+					var error = handleError(failureCodes.PR_ERROR_2.code, err.error);
+					return reject(error);
```

Comment:
change this to `return resolve`

Reply:
This wasn't fixed and code was merged. Please create another PR with the changes requested in this PR.